eng/packages/http-client-csharp/generator/Azure.Generator/src/Primitives/NewAzureProjectScaffolding.cs

```diff
@@ -29,42 +29,102 @@ protected override string GetSourceProjectFileContent()
                 PackageTags = AzureClientPlugin.Instance.Configuration.RootNamespace,
                 GenerateDocumentationFile = true,
             };
-            foreach (var packages in _unbrandedDependencyPackages)
+
+            foreach (var packages in _azureDependencyPackages)
             {
                 builder.PackageReferences.Add(packages);
             }
 
+            int pathSegmentCount = GetPathSegmentCount();
             if (AzureClientPlugin.Instance.InputLibrary.InputNamespace.Auth.ApiKey is not null)
             {
-                ReadOnlySpan<char> text = AzureClientPlugin.Instance.Configuration.OutputDirectory.AsSpan();
-                // we are either a spector project in the eng folder or a real sdk in the sdk folder
-                int beginning = text.IndexOf("eng");
-                if (beginning == -1)
-                {
-                    beginning = text.IndexOf("sdk");
-                }
-                text = text.Slice(beginning);
-                // starting with 2 to include eng at the beginning and src at the end
-                int pathSegmentCount = 2 + text.Count('/');
-                // count both path separators to normalize
-                pathSegmentCount += text.Count('\\');
                 builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude("AzureKeyCredentialPolicy.cs", pathSegmentCount), "Shared/Core"));
+            }
+
+            TraverseInput(out bool hasOperation, out bool hasLongRunningOperation);
+            if (hasOperation)
+            {
                 builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude("RawRequestUriBuilder.cs", pathSegmentCount), "Shared/Core"));
             }
 
+            if (hasLongRunningOperation)
+            {
+                builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude("AppContextSwitchHelper.cs", pathSegmentCount), "Shared/Core"));
```

Comment:
nit: it would make this a lot more concise if we use a for loop and an Enumerable that yields back the different file names.